TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/EditorTests/PropertyTests/IntegerPropertyTests.cs

```diff
@@ -0,0 +1,62 @@
+using System.Collections;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace UnityEditor.ShaderFoundry.UnitTests
+{
+    class IntegerPropertyTests : BlockTestRenderer
+    {
+        IntegerPropertyBlockBuilder BuildWithoutNameOverrides(string defaultValue)
+        {
+            var propBuilder = new IntegerPropertyBlockBuilder
+            {
+                PropertyAttribute = new PropertyAttributeData { DefaultValue = defaultValue },
+                FieldName = "IntField",
+            };
+            return propBuilder;
+        }
+
+        IntegerPropertyBlockBuilder BuildWithNameOverrides(string defaultValue)
+        {
+            var propBuilder = new IntegerPropertyBlockBuilder
+            {
+                PropertyAttribute = new PropertyAttributeData
+                {
+                    UniformName = "_Value",
+                    DisplayName = "Value",
+                    DefaultValue = defaultValue
+                },
+                FieldName = "IntField",
+            };
+            return propBuilder;
+        }
+
+        [UnityTest]
+        public IEnumerator IntegerProperty_DefaultPropertyValueUsed_IsExpectedColor()
+        {
+            var expectedColor = new Color(1, 1, 1, 1);
+
+            var container = CreateContainer();
+            var propBuilder = BuildWithoutNameOverrides("1");
+            var block = propBuilder.Build(container);
+
+            TestSurfaceBlockIsConstantColor(container, propBuilder.BlockName, block, expectedColor);
+            yield break;
+        }
+
+        [UnityTest]
+        public IEnumerator IntegerProperty_MaterialColorSet_IsExpectedColor()
+        {
+            var inputValue = 1;
+            var expectedColor = new Color(inputValue, inputValue, inputValue, 1);
```

Comment:
I suppose it doesn't really matter, but the Float test is checking only .x and here we broadcast to .xyz. Is this intentional?

Reply:
Hm... good catch. I probably copied this one from bool whereas float I copied from a vector type. I'll think about what to do here.